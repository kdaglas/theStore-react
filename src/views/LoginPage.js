import React, { Component } from "react";
import { connect } from "react-redux";
import Login from "../components/loginForm";
import loginAction from "../actions/loginAction";

export class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      attendant_name: "",
      password: "",
      message: ""
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.message) {
        this.setState({
          message: nextProps.message
        });
        nextProps.history.push("/");
      }
    else if (nextProps.error) {
      this.setState({
        message: nextProps.error
      });
    }
  }

  onChange(e) {
    e.preventDefault();
    const { target } = e;
    const { value } = target;
    const { name } = target;
    this.setState({
      [name]: value
    });
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.loginAction(this.state);
  }

  render() {
    return (
      <div>
        <Login
          onChange={this.onChange}
          onSubmit={this.onSubmit}
          message={this.state.message}
          loading={this.state.loading}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  successState: state.authReducer.isSuccessful,
  error: state.authReducer.error,
  message: state.authReducer.message,
  token: state.authReducer.token
});

export default connect(
  mapStateToProps,
  { loginAction }
)(LoginPage);
