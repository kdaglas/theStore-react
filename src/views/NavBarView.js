import React, { Component } from "react";
import { connect } from "react-redux";
import Navbar from "../components/Navbar";
// import Login from "../components/loginForm";
// import loginAction from "../actions/loginAction";

export class NavBarView extends Component {
  state = {
    isAuthenticated: false
  };

  componentWillMount() {
    const token = localStorage.getItem("token");
    if (token) {
      this.setState({
        isAuthenticated: true
      });
    }
  }

  logOut = () => {
    localStorage.removeItem("token");
    window.location = "/";
  };

  render() {
    return (
      <Navbar
        isAuthenticated={this.state.isAuthenticated}
        logOut={this.logOut}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.Auth
  };
}

export default connect(
  mapStateToProps,
  {}
)(NavBarView);
