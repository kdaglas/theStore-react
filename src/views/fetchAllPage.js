import React, { Component } from "react";
import { connect } from "react-redux";
import FetchAll from "../components/fetchAllTable";
import fetchAllAction from "../actions/fetchAllAction";

export class fetchAllPage extends Component {
  state = {};

  componentWillMount = () => {
    const token = localStorage.getItem("token");
    if (!token) {
      window.location.replace("/login");
    }
    this.props.fetchAllAction();
  };
 

  render() {
    return (
      <div>
        <FetchAll products={this.props.products.products} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    products: state.fetchAllReducer
  };
};

export default connect(
  mapStateToProps,
  { fetchAllAction }
)(fetchAllPage);
