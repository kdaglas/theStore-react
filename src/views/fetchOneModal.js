import React, { Component } from "react";
import { connect } from "react-redux";
// import FetchOne from "../components/fetchOne";
// import FetchOne from '../components/fetchAllTable';
import fetchOneAction from "../actions/fetchOneAction";

export class fetchOneModal extends Component {
  state = {};

  componentWillMount = () => {
    const token = localStorage.getItem("token");
    if (!token) {
      window.location.replace("/login");
    }
    this.props.fetchOneAction();
  };
 

  render() {
    return (
      <div>
        <FetchOne product={this.props.product.product} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    product: state.fetchOneReducer
  };
};

export default connect(
  mapStateToProps,
  { fetchOneAction }
)(fetchOneModal);
