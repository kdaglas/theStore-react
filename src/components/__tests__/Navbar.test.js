import React from "react";
import { shallow } from "enzyme";
import { Redirect } from "react-router-dom";
import {NavBarView } from "../../views/NavBarView";

describe("the navigation bar Component", () => {
  it("should shore navigation bar without crushing", () => {
    const component = shallow(<NavBarView />);
    expect(component).toMatchSnapshot();
  });

  it("should log out when clicked", () => {
    const token = localStorage.setItem('token', 'token')
    const props = {}
    const wrapper = shallow(<NavBarView {...props} />);
    wrapper.instance().logOut()
    expect(localStorage.getItem('token')).toBeNull()
   
  });
});
