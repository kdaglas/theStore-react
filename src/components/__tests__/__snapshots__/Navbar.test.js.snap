// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the navigation bar Component should shore navigation bar without crushing 1`] = `ShallowWrapper {}`;
